packages/code-du-travail-api/src/server/routes/items/index.js

```diff
@@ -35,12 +37,42 @@ router.get("/items/:source/:slug", async ctx => {
   const item = response.body.hits.hits[0];
 
   // Get current item title to find related items
-  const { title } = item._source;
-  const relatedItems = await getSearch({
-    q: title,
-    excludeSources: `${SOURCES.CDT}, ${SOURCES.CCN}, ${SOURCES.THEMES}`
+  const {
+    _id,
+    _source: { title_vector: query_vector }
+  } = item;
+  const size = MAX_RESULTS;
+  const sources = [
+    SOURCES.TOOLS,
+    SOURCES.SHEET_SP,
+    SOURCES.SHEET_MT,
+    SOURCES.LETTERS,
+    SOURCES.CONTRIBUTIONS
+  ];
+  const {
+    body: {
+      responses: [esResponse, semResponse]
+    }
+  } = await elasticsearchClient.msearch({
+    body: [
+      { index },
+      { ...getRelatedItemsBody({ index, id: _id, size, sources }) },
+      { index },
+      // we +1 the size to remove the document source that should match perfectly for the given vector
+      { ...getSemBody({ query_vector, size: size + 1, sources }) }
+    ]
   });
 
+  const { hits: { hits: semanticHits } = { hits: [] } } = semResponse;
+  const { hits: { hits: bem25Hits } = { hits: [] } } = esResponse;
+
+  const relatedItems = utils
+    .mergePipe(
+      semanticHits,
+      bem25Hits.filter(doc => doc._id !== _id),
```

Comment:
shouldn't the filter also apply to semanticHits ?

Reply:
oh good cathc it is the contratry filter for semantic results not bem25 since they provide from a more_like_this query